### PR TITLE
Fix up some tsp naming

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Cleaned up client names.
Include Begin prefix for LRO method options type.

Fixes https://github.com/Azure/autorest.go/issues/1207

### CODE REVIEW

File: packages/typespec-go/test/armapicenter/fake/zz_apidefinitions_server.go [hunk 1]
```diff
@@ -18,60 +18,60 @@ import (
 	"regexp"
 )
 
-// ApiDefinitionsServer is a fake server for instances of the armapicenter.ApiDefinitionsClient type.
-type ApiDefinitionsServer struct {
-	// CreateOrUpdate is the fake for method ApiDefinitionsClient.CreateOrUpdate
+// APIDefinitionsServer is a fake server for instances of the armapicenter.APIDefinitionsClient type.
+type APIDefinitionsServer struct {
```
Comment (RickWinter) [inline packages/typespec-go/test/armapicenter/fake/zz_apidefinitions_server.go:22]: Do we capitalize `API` everywhere? API almost always is followed by a capitilized object. Thus API almost always shows up as APIA, APIB, APIC, APID, .....

File: packages/typespec-go/test/cadlranch/special-headers/clientreqidgroup/custom_client.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 )
 
-func NewClientRequestIdClient(options *azcore.ClientOptions) (*ClientRequestIdClient, error) {
+func NewClientRequestIDClient(options *azcore.ClientOptions) (*ClientRequestIDClient, error) {
```
Comment (RickWinter) [inline packages/typespec-go/test/cadlranch/special-headers/clientreqidgroup/custom_client.go:17]: Ditto about API when it comes to `ID`. Is `ID` always capitilized in GO?

**Review by RickWinter — APPROVED** [review]

File: packages/typespec-go/test/cadlranch/special-headers/clientreqidgroup/custom_client.go (hunk 2 above)
Comment (jhendrixMSFT) [reply packages/typespec-go/test/cadlranch/special-headers/clientreqidgroup/custom_client.go:17]: Acronyms are always capitalized in Go and we have a list in the code generator that `ensureNameCase()` uses in order to perform the capitalization.

File: packages/typespec-go/test/armapicenter/fake/zz_apidefinitions_server.go (hunk 1 above)
Comment (jhendrixMSFT) [reply packages/typespec-go/test/armapicenter/fake/zz_apidefinitions_server.go:22]: Yes, acronyms are always capitalized.

**Pull request merged**